heron/instance/src/java/com/twitter/heron/instance/bolt/BoltInstance.java

```diff
@@ -192,8 +192,8 @@ public void readTuplesAndExecute(Communicator<HeronTuples.HeronTupleSet> inQueue
       int nValues = topologyContext.getComponentOutputFields(
           stream.getComponentName(), stream.getId()).size();
 
-      // We would reuse the System.nanoTime()
-      long currentTime = startOfCycle;
+      // Record current time at the start of processing a set of tuples
+      long currentTime = System.nanoTime();
```

Comment:
Can you clarigy (in comments) what `currentTime` is expected to reflect when it's used after the loop? It's not clear from reading the code.